Add CI test using Github Actions

Jialn opened the pull request:
Added build and code checking for playground_test. 

The script installs gazebo dependency every time. To avoid this I also build a docker image https://hub.docker.com/layers/jialn/socialbot/latest/images/sha256-6db01ed73b6516995a4dbf971d4228a53baf1f0c00a91213fea1be468c6defca

But it seems that installing gazebo dependency (need to get some 200~400MB from Azure internal sources) is fast enough, and maybe faster than pulling image from docker-hub (~800MB). The setup and installing progress take about 2.5 minutes.

emailweixu commented on .github/workflows/python_test.yml:
```diff
@@ -0,0 +1,40 @@
+    - name: Lint with flake8
+      run: |
+        pip install flake8
+        # distable static check for now
```
distable => disable

Jialn replied:
Fixed

emailweixu commented on .github/workflows/python_test.yml:
```diff
@@ -0,0 +1,40 @@
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip3 install -e .
```
This is a duplicate of line 29.

Jialn replied:
the first one is needed for cmake and compiling, another one to setup pygazebo after compiling.

emailweixu commented on .github/workflows/python_test.yml:
```diff
@@ -0,0 +1,40 @@
+      run: |
+        python -m pip install --upgrade pip
+        pip3 install -e .
+        pip3 install absl-py pytest Pillow
```
should add these dependencies into setup.py instead.

Jialn replied:
Done

emailweixu commented on .github/workflows/python_test.yml:
```diff
@@ -0,0 +1,38 @@
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 ./python --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
It's better to follow https://github.com/HorizonRobotics/alf/blob/master/.ci-cd/build.sh to use the same style check to be consistent with ALF style.

Jialn replied:
Added the the same style check using pre-commit. 
This flake8 checking can be kept, could do static checking for syntax errors or undefined names in corner case that tests does not cover, also it can report warnings like unused import, variables assigned to but never used. max-line-length checking can be ignored since pre-commit already checked that.

witwolf commented on .github/workflows/python_test.yml:
```diff
@@ -0,0 +1,40 @@
+        # flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with pytest
+      run: |
+        GAZEBO_MODEL_PATH=./python/social_bot/models xvfb-run python python/social_bot/envs/play_ground_test.py
```
Is it possible to run all testes by `GAZEBO_MODEL_PATH=./python/social_bot/models xvfb-run python -m unittest discover -p "*_test.py" -v` ?

Jialn commented on .github/workflows/python_test.yml:
```diff
@@ -0,0 +1,48 @@
+        fi
+    - name: Test with pytest
+      run: |
+        cd ./python/social_bot/envs
```
"unittest discover" need to run in this path. Let's only run the tests under env for now.

Jialn commented:
https://github.com/HorizonRobotics/SocialRobot/issues/102 This issue generates lots of warnings about textures during the tests. I've removed unused texture and stl files in our customized models, and added the original models to our repo because of the texture dependency.
